test/verify/check-connection

```diff
@@ -135,14 +135,21 @@ class TestConnection(testlib.MachineCase):
         b.enter_page("/system")
 
         # sever the connection on the server page
-        m.execute("iptables -w -I INPUT -p tcp --dport 9090 -j REJECT")
-        b.switch_to_top()
-        with b.wait_timeout(60):
-            b.wait_visible(".curtains-ct")
+        # would be nice to use `firewall-cmd --add-rich-rule`, but firewalld always allows "established" connections
+        m.execute("nft add table ip cockpittest")
+        m.execute("nft add chain ip cockpittest INPUT '{ type filter hook input priority 0; policy accept; }'")
+        m.execute("nft insert rule ip cockpittest INPUT tcp dport 9090 reject")
+
+        try:
+            b.switch_to_top()
+            with b.wait_timeout(60):
+                b.wait_visible(".curtains-ct")
+
+            b.wait_in_text(".curtains-ct h1", "Disconnected")
+            b.wait_in_text('.curtains-ct .pf-v5-c-empty-state__body', "Connection has timed out.")
+        finally:
+            m.execute("nft delete table ip cockpittest")
```

Comment:
I guess this is basically something like:
 - I want a reliable cleanup because this test is non-destructive
 - but I don't want it at the end of the function because I'm going to do some more stuff after this...
 - I don't have a reasonable way to cause the cleanup to run earlier (and remove it from the list to run at the end of the test)

The pattern looks unusual and because of it it's a bit difficult to understand the intent.  I wonder if we could invent some new idiom to do a better job about that...

Reply:
for the record, I was imagining something like:

```python
with testlib.cleanup(m.execute, "nft delete table ip cockpittest"):
    m.execute("nft add table ip cockpittest")
    ...
```

Reply:
It's not that unusual: check-{connection,networkmanager-firewall,packagekit,system-info,system-services} etc. all do that already. I thought try/finally was *the* Pythonic pattern for that. There's generators too, but writing one is quite some overhead. Do you have something in mind?

Reply:
We could also make the existing `addCleanup` function return a context manager which does the current behaviour if unused, but otherwise on `__exit()__` performs the cleanup operation directly (and removes itself from the list)

Reply:
[addCleanup()](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup) by itself doesn't return a context manager. There's an [enterContext()](https://docs.python.org/3/library/unittest.html#unittest.TestCase.enterContext) which I've never seen/used, and seems to be related somehow, but there's no examples how to put these together..

Writing our own CM would certainly work, but I'm not really convinced that it is better? The code becomes "upside down" (cleanup first), and it's more custom code to carry around and teach to developers.

Anyway, I'd like to land this to unravel the c10s story. If you feel bored, there's plenty of existing try/finally to clean up already :grin: